File: 프로그래머스/0/120862. 최댓값 만들기 （2）/최댓값 만들기 （2）.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> numbers) {
    sort(numbers.begin(), numbers.end()); // 오름차순 정렬
    
    if(numbers[0] < 0 && numbers[1] < 0){
        int tmpMax1 = numbers[0] * numbers[1];
        int tmpMax2 = numbers[numbers.size()-2] * numbers[numbers.size()-1];
        
        if(tmpMax1 > tmpMax2) return tmpMax1;
        else return tmpMax2;
    }
    
    return numbers[numbers.size()-2] * numbers[numbers.size()-1];
}

File: 프로그래머스/0/120862. 최댓값 만들기 （2）/최댓값 만들기 （2）.py
def solution(numbers):
    answer = 0
    numbers.sort()
    
    if numbers[0] < 0 and numbers[1] < 0:
        tmpMax1 = numbers[0] * numbers[1]
        tmpMax2 = numbers[-1] * numbers[-2]
        if tmpMax1 > tmpMax2:
            return tmpMax1
    
    return numbers[-1] * numbers[-2]